apps/mobile/src/features/approver/layouts/base-stx-message-approver.layout.tsx

```diff
@@ -0,0 +1,155 @@
+import { Divider } from '@/components/divider';
+import { ClarityValueListDisplayer } from '@/features/approver/clarity-value.section';
+import { ApproverAccountCard } from '@/features/approver/components/approver-account-card';
+import { Account } from '@/store/accounts/accounts';
+import { makeAccountIdentifer } from '@/store/utils';
+import { t } from '@lingui/macro';
+import { ChainId } from '@stacks/network';
+import {
+  ClarityType,
+  ClarityValue,
+  StringAsciiCV,
+  TupleCV,
+  UIntCV,
+  cvToString,
+  deserializeCV,
+} from '@stacks/transactions';
+
+import { stacksChainIdToCoreNetworkMode } from '@leather.io/stacks';
+import { Approver, Box, Button, Cell, Text } from '@leather.io/ui/native';
+import { capitalize } from '@leather.io/utils';
+
+interface BaseStxMessageApproverLayoutProps {
+  onApprove(): void;
+  onCloseApprover(): void;
+  accountId: string | null;
+  accounts: Account[];
+  messageToSign:
+    | {
+        messageType: 'utf8';
+        message: string;
+      }
+    | {
+        messageType: 'structured';
+        message: string;
+        domain: string;
+      };
+}
+function cvToDisplay(cv: ClarityValue): string {
+  return cvToString(cv).replaceAll('"', '');
+}
+function chainIdToDisplay(chainIdCv: ClarityValue): string {
+  if (chainIdCv.type !== ClarityType.UInt) return '';
+  const chainIdString = cvToString(chainIdCv);
+  const chainId = parseInt(chainIdString.replace('u', ''), 10);
+  if (!Object.values(ChainId).includes(chainId)) return '';
+  return capitalize(stacksChainIdToCoreNetworkMode(chainId));
+}
+
+export type StructuredMessageDataDomain = TupleCV<{
+  name: StringAsciiCV;
+  version: StringAsciiCV;
+  'chain-id': UIntCV;
+}>;
+
+function Structured({
```

Comment:
Maybe we can:
- name this **StructuredMessage** or something with more info?
- give it an interface type 
- move it to it's own file?